FWCore/ParameterSet/python/Config.py

```diff
@@ -2136,11 +2184,41 @@ def __init__(self, **kargs):
                 ), **kargs)
     specialImportRegistry.registerSpecialImportForType(SwitchProducerTest2, "from test import SwitchProducerTest2")
 
+    class TestModuleTypeResolver:
+        def __init__(self, accelerators, pset=untracked.PSet()):
+            # first element is used as the default is nothing is set
+            self._valid_backends = []
+            if "test1" in accelerators:
+                self._valid_backends.append("test1_backend")
+            if "test2" in accelerators:
+                self._valid_backends.append("test2_backend")
+            if len(self._valid_backends) == 0:
+                raise EDMException(edm.errors.UnavailableAccelerator, "Machine has no accelerators that Test supports (has {})".format(", ".join(accelerators)))
+            self._pset = pset
+
+        def pluginAndConfiguration(self):
+            return ("TestModuleTypeResolver", self._pset)
+
+        def setModuleBackend(self, module):
```

Comment:
I think this member function should be renamed. `setModuleVariant()` comes to my mind. Any better ideas?

Reply:
Now named as `setModuleVariant()` (better suggestions are still welcome)